core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMapFactory.java

```diff
@@ -140,6 +142,11 @@ public abstract DataMapBuilder createBuilder(Segment segment, String shardName,
    */
   public abstract void deleteDatamapData();
 
+  /**
+   * delete datamap data if any
+   */
+  public abstract void deleteSegmentDatamapData(String segmentNo) throws IOException;
```

Comment:
Can u add default implementation here so all the child classes which ever want their specific implementation can override, currently all the child class implementing this method without any logic

Reply:
done